infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -91,15 +93,35 @@ var ValidOfflineStoreFilePersistenceTypes = []string{
 	"duckdb",
 }
 
+// OfflineStoreDBStorePersistence configures the DB store persistence for the offline store service
+type OfflineStoreDBStorePersistence struct {
+	// +kubebuilder:validation:Enum=snowflake.offline;bigquery;redshift;spark;postgres;feast_trino.trino.TrinoOfflineStore;redis
+	Type          string                       `json:"type,omitempty"`
+	SecretRef     *corev1.LocalObjectReference `json:"secretRef,omitempty"`
```

Comment:
should these be required fields?
```suggestion
	Type          string                       `json:"type"`
	SecretRef     *corev1.LocalObjectReference `json:"secretRef"`
```

Reply:
@dmartinol gave a thumbs up. @tmihalac from a code & usability standpoint... aren't these 2 fields required if the user decides to configure a db store? (i believe they are) if so, we need to make them required in the CRD.

Reply:
Tracked new GH issue with leftovers